webhooks/core/service_mutator.go

```diff
@@ -52,7 +52,41 @@ func (m *serviceMutator) MutateCreate(ctx context.Context, obj runtime.Object) (
 }
 
 func (m *serviceMutator) MutateUpdate(ctx context.Context, obj runtime.Object, oldObj runtime.Object) (runtime.Object, error) {
-	return obj, nil
+	// this mutator only cares about Service objects
+	newSvc, ok := obj.(*corev1.Service)
+	if !ok {
+		return obj, nil
+	}
+
+	oldSvc, ok := oldObj.(*corev1.Service)
+	if !ok {
+		return obj, nil
+	}
+
+	if newSvc.Spec.Type != corev1.ServiceTypeLoadBalancer {
+		return obj, nil
+	}
+
+	// does the old Service object have spec.loadBalancerClass?
+	if oldSvc.Spec.LoadBalancerClass != nil && *oldSvc.Spec.LoadBalancerClass != "" {
+		// if so, let's inspect the incoming object for the same field
+
+		// does the new Service object lack spec.loadBalancerClass?
+		// if so, set it to the old value
+		// if yes, then leave it be because someone wanted it that way, let the user deal with the error
+		if newSvc.Spec.LoadBalancerClass == nil || *newSvc.Spec.LoadBalancerClass == "" {
+			newSvc.Spec.LoadBalancerClass = oldSvc.Spec.LoadBalancerClass
+
```

Comment:
I was just thinking some cornor cases - would this change cause some issue when this field in the newSvc is intended to be nil or ""?

Reply:
Since this handler is for updates to an _existing_ service only, there's a couple of reasons the field would be missing/empty:

1. User is updating the service object via `patch`, submitting only modified fields. In this case, tacking the LBC field onto the update is a noop, and there's no change in behavior.
2. User is updating the service object via `put` (this is the `kubectl replace`) and omits the field. The normal response would be an error, as the field is immutable.
With this handler, the error does not occur as we putting the field back. This is a change in behavior.
Whether this change is unexpected - I think it is not, because presumably user has enabled the webhook, is aware of the webhook effects, and expects LB Services in their cluster to be managed by AWS LBC. So this change, then, just prevents the user from running into an error (and this is the entire reason behind this PR because our deployment tooling - Helm - runs into this very error).
Importantly, because the handler inspects the existing Service, it will only touch service that are known to be managed by AWS LBC, and not something else.